docs/en/operation/Admin-CLI.md

```diff
@@ -362,3 +396,158 @@ alluxio.master.audit.logging.enabled
+This command returns either when all workers are idle, or when this wait time is up. The default value is `5m`.
+
+`--disable/-d` is an optional argument. If this is specified, the decommissioned workers will not be able to
+register to the master. In other words, a disabled worker cannot join the cluster and will not be chosen for I/O requests.
+This is useful when the admin wants to remove the workers from the cluster. When those disabled workers register,
+the master will reject them but will not kill the worker processes. This is often used in pair with the `enableWorker` command.
+
+The command will perform the following actions:
+
+1. For each worker in the batch, send a decommission command to the primary Alluxio master so 
+   the master marks those workers as decommissioned and will not serve operations. The ongoing I/O on those workers
+   will NOT be interrupted.
+2. It takes a small interval for all other Alluxio components (like clients and Proxy instances)
+   to know those workers should not be used, so this command waits for the interval time defined by `alluxio.user.worker.list.refresh.interval`
+   on the node where this command is executed. Before a client/proxy realizes the workers are decommissioned,
+   they may submit more I/O requests to those workers, and those requests should execute normally.
+3. Get the active worker list from the master after waiting, and verify the target workers are not active anymore.
+4. Wait for the workers to become idle. This command will constantly check the idleness status on each worker.
+5. Either all workers have become idle, or the specified timeout expires, this command will return.
+
+**Limitations**
+
+In some cases, the `decommissionWorker` command may return code 0 (success) but when a worker process is killed,
+some user I/O may fail.
+
+When the `decommissionWorker` command waits for a worker to become idle, it only respects ongoing I/O requests on the worker.
+If the Alluxio client is reading/writing the worker with short circuit, the client directly reads/writes cache files
+in worker storage and maintains a gRPC stream with the worker simply for locking that block. An open(and idle) stream
+will NOT be respected by the `decommissionWorker` command so it may consider the worker is idle.
+However, when the admin kills the worker and deletes the cache storage, the client request will fail
+either because the cache blocks are gone or because the gRPC stream is broken. So if you are using short circuit,
+wait a few more minutes before killing the worker process and deleting cached blocks. The clients will
+know the workers are decommissioned (should not be read/written) and stop using those workers by short circuit.
+
+The `decommissionWorker` command does NOT consider cache blocks on the target workers. That means if
+decommissioning some workers will bring ALL replicas of certain blocks offline, and those blocks only exist in cache,
```

Comment:
```suggestion
there is a block whole sole replica happens to reside on one of the workers that will be decommissioned,
and it's not in UFS either (due to `MUST_CACHE` or `ASYNC_THROUGH` write mode,
```

Reply:
It can also be the case that one block has multiple replicas, and all replicas are on this batch of workers.